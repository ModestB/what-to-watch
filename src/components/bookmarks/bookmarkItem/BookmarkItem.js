import React from 'react';

import classes from './BookmarkItem.module.scss';
import CloseIcon from '../../../icons/js/Close';

const BookmarkItem  = React.memo(function BookmarkItem (props) {
  function clickHandler () {
    props.displayBookmarksHandler();
    props.findShowById(props.id, props.mediaType);
  }

  return (
    <div className='d-flex justify-content-between align-items-center pb-2'>
      <div 
        className={`${classes.title} d-flex text-left`}
        onClick={clickHandler}
      >
        {`${props.title} (${props.date})`}
      </div>
      
      <CloseIcon 
        className={`${classes.closeIcon}`}
        onClick={() => props.removeBookmark(props.id)}
        width="20px" 
        height="20px"
      />
    </div>
  )
});

export default BookmarkItem;