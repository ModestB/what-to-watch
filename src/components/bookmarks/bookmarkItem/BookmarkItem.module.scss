@import 'assets/styles/variables';

.container{
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 8px;
}

.title{
  font-size: 16px;
  text-align: left;
  line-height: 1.3;
  &:hover{
    cursor: pointer;
    color: $color--primary;
  }
}