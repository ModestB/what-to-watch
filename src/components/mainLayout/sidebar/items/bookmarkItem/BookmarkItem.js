import React, { useEffect, useState, useRef } from "react";
import { connect } from "react-redux";

import classes from "./BookmarkItem.module.scss";

import HeartIcon from "../../../../../icons/js/Heart";
import SidebarItem from "../SidebarItem";

export const BookmarksItem = (props) => {
  const isInitialMount = useRef(true);
  let [heartKey, setHeartKey] = useState(0);
  useEffect(() => {
    if (isInitialMount.current) {
      isInitialMount.current = false;
    } else {
      setHeartKey(heartKey + 1);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.bookmarks]);

  return (
    <SidebarItem title="Bookmarks" route="bookmarks">
      <HeartIcon
        key={heartKey > 1 ? heartKey : "0"}
        fill="#ffffff"
        width="30px"
        height="30px"
        className={classes.iconHeart}
      />
    </SidebarItem>
  );
};

const mapStateProps = (state) => {
  return {
    bookmarks: state.bookmarks,
  };
};

export default connect(mapStateProps)(BookmarksItem);
