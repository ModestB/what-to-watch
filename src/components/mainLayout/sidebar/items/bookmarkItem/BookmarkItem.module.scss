@import "assets/styles/variables";
@import "assets/styles/keyframes";

.counter {
  padding: 0 $spacing-1;
}

.iconHeart {
  animation: scale-up-center 0.3s ease-in-out 2 alternate both;
}
