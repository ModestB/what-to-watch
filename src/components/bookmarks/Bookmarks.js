import React, { useEffect, useState, useRef } from "react";
import { connect } from "react-redux";

// Action Types
import { toggleBookmarks, removeBookmark } from "../../store/actions/actions";

import classes from "./Bookmarks.module.scss";

import BookmarkItem from "./bookmarkItem/BookmarkItem";
import HeartIcon from "../../icons/js/Heart";
import CloseIcon from "../../icons/js/Close";

export const Bookmarks = (props) => {
  const isInitialMount = useRef(true);
  let [heartKey, setHeartKey] = useState(0);
  useEffect(() => {
    if (isInitialMount.current) {
      isInitialMount.current = false;
    } else {
      setHeartKey(heartKey + 1);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.bookmarks]);

  let bookmarks = <p>No bookmarks</p>;

  if (props.bookmarks && props.bookmarks.length > 0) {
    bookmarks = props.bookmarks.map((bookmark) => {
      return (
        <BookmarkItem
          id={bookmark.id}
          key={bookmark.id}
          title={bookmark.title}
          mediaType={bookmark.mediaType}
          date={bookmark.date}
        />
      );
    });
  }
  return (
    <div
      className={`
        ${classes.container} 
        ${props.displayBookmarks ? classes.active : ""}
      `}
    >
      <header className={`${classes.header}`} onClick={props.toggleBookmarks}>
        <div className={`${classes.header__wrp}`}>
          <p className={`${classes.title}`}> To Watch</p>
          <HeartIcon
            key={heartKey > 1 ? heartKey : "0"}
            fill="#ffffff"
            width="30px"
            height="30px"
            className={classes.iconHeart}
          />
          <span className={`${classes.counter}`}>
            {props.bookmarks ? props.bookmarks.length : 0}
          </span>
        </div>

        <CloseIcon
          className={`${classes.iconClose} icon-close icon-close--light`}
          width="20px"
          height="20px"
        />
      </header>
      <div className={`${classes.body} customScroll customScroll--red`}>
        {bookmarks}
      </div>
    </div>
  );
};

const mapStateProps = (state) => {
  return {
    displayBookmarks: state.displayBookmarks,
    bookmarks: state.bookmarks,
  };
};

const mapStateDispatch = (dispatch) => {
  return {
    toggleBookmarks: () => dispatch(toggleBookmarks()),
    removeBookmark: (bookmarkId) => dispatch(removeBookmark(bookmarkId)),
  };
};

export default connect(mapStateProps, mapStateDispatch)(Bookmarks);
