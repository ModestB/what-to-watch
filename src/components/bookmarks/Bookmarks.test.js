import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { Bookmarks } from "./Bookmarks";
import { BookmarkItem } from "./bookmarkItem/BookmarkItem";

configure({ adapter: new Adapter() });

describe("<Bookmarks />", () => {
  let wrapper;
  let bookmarks = [
    {
      id: 420809,
      title: "Maleficent: Mistress of Evil",
      date: "2019",
      mediaType: "movie",
    },
    {
      id: 69823,
      title: "After M*A*S*H",
      date: "1983",
      mediaType: "tv",
    },
    {
      id: 49434,
      title: "Making M*A*S*H",
      date: "Not specified",
      mediaType: "tv",
    },
  ];

  beforeEach(() => {
    wrapper = shallow(<Bookmarks />);
  });

  it("should render initial layout no bookmarks", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should render initial layout with bookmarks", () => {
    wrapper.setProps({ bookmarks: bookmarks });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should render 3 bookmarks/.body has 3 childs", () => {
    wrapper.setProps({ bookmarks: bookmarks });
    expect(wrapper.find(".body").children()).toHaveLength(3);
  });

  it("should render 0 bookmarks/.body has 1 child", () => {
    expect(wrapper.find(".body").children()).toHaveLength(1);
    expect(wrapper.find(".body p")).toHaveLength(1);
  });
});
