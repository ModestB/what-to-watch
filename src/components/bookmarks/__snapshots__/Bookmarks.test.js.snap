// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bookmarks /> should render initial layout no bookmarks 1`] = `
"<div className=\\"\\\\n        container \\\\n        \\\\n      \\">
  <header className=\\"header\\" onClick={[undefined]}>
    <div className=\\"header__wrp\\">
      <p className=\\"title\\">
         To Watch
      </p>
      <SvgHeart fill=\\"#ffffff\\" width=\\"30px\\" height=\\"30px\\" className=\\"iconHeart\\" />
      <span className=\\"counter\\">
        0
      </span>
    </div>
    <SvgClose className=\\"iconClose icon-close icon-close--light\\" width=\\"20px\\" height=\\"20px\\" />
  </header>
  <div className=\\"body customScroll customScroll--red\\">
    <p>
      No bookmarks
    </p>
  </div>
</div>"
`;

exports[`<Bookmarks /> should render initial layout with bookmarks 1`] = `
"<div className=\\"\\\\n        container \\\\n        \\\\n      \\">
  <header className=\\"header\\" onClick={[undefined]}>
    <div className=\\"header__wrp\\">
      <p className=\\"title\\">
         To Watch
      </p>
      <SvgHeart fill=\\"#ffffff\\" width=\\"30px\\" height=\\"30px\\" className=\\"iconHeart\\" />
      <span className=\\"counter\\">
        3
      </span>
    </div>
    <SvgClose className=\\"iconClose icon-close icon-close--light\\" width=\\"20px\\" height=\\"20px\\" />
  </header>
  <div className=\\"body customScroll customScroll--red\\">
    <Connect(Component) id={420809} title=\\"Maleficent: Mistress of Evil\\" mediaType=\\"movie\\" date=\\"2019\\" />
    <Connect(Component) id={69823} title=\\"After M*A*S*H\\" mediaType=\\"tv\\" date=\\"1983\\" />
    <Connect(Component) id={49434} title=\\"Making M*A*S*H\\" mediaType=\\"tv\\" date=\\"Not specified\\" />
  </div>
</div>"
`;
